utils/monitoring_tools/pytorch_model_internals/README.md

```diff
@@ -0,0 +1,252 @@
+# TorchWatcher
+
+A lightweight PyTorch model monitoring tool that automatically tracks layer activations, gradients, and parameters during training. Built for seamless integration with neptune.ai.
+
+## Why TorchWatcher?
+
+Training deep learning models often involves monitoring internal layer behavior to understand model performance, debug issues, and optimize training. However, manually implementing hooks and logging for each layer can be time-consuming and error-prone. TorchWatcher solves this by providing:
+
+- **Zero-configuration monitoring**: Automatically detects and tracks all layers in your PyTorch model
+- **Comprehensive insights**: Monitor activations, gradients, and parameters with built-in statistical analysis
+- **Seamless Neptune integration**: Direct logging to Neptune with organized namespacing for easy experiment tracking
+- **Minimal performance impact**: Optimized for production use with configurable tracking options
+- **Memory efficient**: Automatically cleans up tensors to prevent memory leaks during long training runs
+
+Whether you're debugging gradient flow issues, monitoring activation patterns, or tracking parameter changes during training, TorchWatcher provides the insights you need without the overhead of manual implementation.
+
+## Features
+
+- **Automatic Layer Tracking**: Monitors activations, gradients, and parameters of specified PyTorch layers
+- **Flexible Layer Selection**: Track all layers or specify which layer types to monitor
+- **Comprehensive Statistics**: Predefined tensor statistics including mean, standard deviation, norm, min, max, variance, and absolute mean
+- **Configurable Tracking**: Enable/disable tracking of activations, gradients, and parameters as needed
+- **Organized Logging**: Structured metric namespacing for better organization in Neptune
+- **Memory Efficient**: Clears stored tensors after each logging step
+- **Error Handling**: Robust error handling with informative warnings
+
+## Changelog
+
+### [v0.1.0] - 2024-03-19
+
+#### Added
+- Initial release of TorchWatcher
+
+## Prerequisites
+
+- Python 3.8 or higher
+- PyTorch 2.0 or higher
+- CUDA 11.7 or higher (for GPU support)
+- Neptune account and API token
+
+## Installation
+
+```bash
+pip install neptune_scale
+```
+
```

Comment:
A bit misleading as it implies that installing `neptune_scale` will also install the `TorchWatcher`.
Having the `neptune_scale` package installed should be mentioned in the prerequisites

```suggestion
```